File: problemas/iniciante/2166/2166.py
N = int(input())

resposta = 0.0
if(N == 0):
    resposta = 1.0
else:
    resposta = 2.0
    for _ in range(N - 1):
        resposta = 2.0 + 1.0 / resposta
    resposta = 1.0 + 1.0 / resposta

print(f'{resposta:.10f}')

File: problemas/iniciante/2166/2166.clj
(let [n (read-string (read-line))
      resposta (atom 0.0)]
  (if (= n 0)
    (reset! resposta 1.0)
    (do
      (reset! resposta 2.0)
      (doseq [_ (range 1 n)]
        (reset! resposta (+ 2.0 (/ 1.0 @resposta))))
      (reset! resposta (+ 1.0 (/ 1.0 @resposta)))))
  (printf "%.10f\n" @resposta))